### DISCUSSION TITLE
OCOMMAND decoding

### DISCUSSION THREAD

**Discussion by justbruce** [esphome-econet/esphome-econet#343] [Ideas]
The OCOMMAND descriptor seems to have the commanded Inverter Crank Speed embedded, as when one divides the value by 65536 (2^16), out comes the inverter crank speed.  There is a remainder, however, and I'm sort of puzzling over what it could be, or if I've gotten the encoding wrong.

For example

`[13:45:55][I][econet:261]:   OCOMMAND: 117977600.000000` ==1800.1953125 - This most often settles at 1800, but can sometimes go up to 2050 RPM for brief periods

I'm trying to figure out if there is a significance to the value being essentially a 16 bit number - what else might be encoded here?

**Comment by Daniel-dev22** [comment]
> The OCOMMAND descriptor seems to have the commanded Inverter Crank Speed embedded, as when one divides the value by 65536 (2^16), out comes the inverter crank speed.  There is a remainder, however, and I'm sort of puzzling over what it could be, or if I've gotten the encoding wrong.
> 
> For example
> 
> `[13:45:55][I][econet:261]:   OCOMMAND: 117977600.000000` ==1800.1953125 - This most often settles at 1800, but can sometimes go up to 2050 RPM for brief periods
> 
> 
> I'm trying to figure out if there is a significance to the value being essentially a 16 bit number - what else might be encoded here?

I think this is used to control the unit from the thermostat. As in how fast to spin etc.

**Comment by justbruce** [comment]
Correct.  I'm trying to figure out if the remainder encodes other commands to the unit.  I think it does, but how it does so [fraction, convert to HEX/octal/binary/???] is my quandary

**Comment by justbruce** [comment]
Additional Log captures of OCOMMAND.

`[econet:261]:   OCOMMAND: 118895104.000000` ==1814.1953125
`[econet:261]:   OCOMMAND: 118698496.000000` ==1811.1953125
`[econet:261]:   OCOMMAND: 118043136.000000` ==1801.1953125

All of these provide the same behavior - speed settles at 1800RPM

`[econet:261]:   OCOMMAND: 128725504.000000` ==1964.1953125

Settles at 1960

`[econet:261]:   OCOMMAND: 128070144.000000` == 1954.1953125

Settles at 1950

**Comment by barndawgie** [comment]
Might be worth opening an Issue with some of the details here.

**Reply by bitfliq** [reply]
Can you post the raw hex command that includes this before it gets split into the variable/value? Looking for 0707f8f8. Specifically wondering if the first two bytes are themselves the crank speed (or maybe it adds on the next two and then divides by 1000).
